kubel.el

```diff
@@ -532,26 +539,29 @@ ARGS is the arguments list from transient."
 (defun kubel-set-namespace ()
   "Set the namespace."
   (interactive)
-  (let ((namespace (completing-read "Namespace: " kubel-namespace-history
-                                    nil nil nil nil "default")))
-    (when (get-buffer (kubel--buffer-name))
-      (kill-buffer (kubel--buffer-name)))
+  (let* ((namespace (completing-read "Namespace: " kubel-namespace-history
+                                     nil nil nil nil "default"))
+	 (kubel--buffer (get-buffer (kubel--buffer-name)))
+	 (last-default-directory (when kubel--buffer
+				   (with-current-buffer kubel--buffer default-directory))))
+    (when kubel--buffer (kill-buffer kubel--buffer))
     (setq kubel-namespace namespace)
     (unless (member namespace kubel-namespace-history)
       (push namespace kubel-namespace-history))
-    (kubel)))
+    (kubel last-default-directory)))
```

Comment:
make sure the new created buffer in 'kubel' having same 'default-directory' with the killed buffer